internal/metrics/metrics.go

```diff
@@ -158,3 +197,34 @@ func AddLabels(current, add prometheus.Labels) prometheus.Labels {
 
 	return returned
 }
+
+// RegisterSpanEvent adds a span event along with the given attributes.
+//
+// event here is arbitrary and can be in the form of strings like - publishCondition, updateCondition etc
+func RegisterSpanEvent(span trace.Span, condition *rctypes.Condition, workerID, serverID, event string) {
```

Comment:
Should this include an error parameter? If not, should we provide another function that takes a span and annotates it with an error (to make correlation in Honeycomb easier)?